tests/baselines/reference/conditionalTypes2.errors.txt

```diff
@@ -32,7 +32,7 @@ tests/cases/conformance/types/conditional/conditionalTypes2.ts(74,12): error TS2
   Property 'bat' is missing in type 'Foo & Bar' but required in type '{ foo: string; bat: string; }'.
 tests/cases/conformance/types/conditional/conditionalTypes2.ts(75,12): error TS2345: Argument of type 'Extract2<T, Foo, Bar>' is not assignable to parameter of type '{ foo: string; bat: string; }'.
   Type 'T extends Bar ? T : never' is not assignable to type '{ foo: string; bat: string; }'.
-    Type 'Bar & Foo & T' is not assignable to type '{ foo: string; bat: string; }'.
+    Property 'bat' is missing in type 'Bar & Foo' but required in type '{ foo: string; bat: string; }'.

```

Comment:
This is really the only test that's affected? We don't have any other tests that try to intersect a type with `{}`?

Reply:
We have plenty of those, what you're seeing here is the difference between normalizing all intersections in relationship checking vs. just normalizing intersections containing `{}`. Previously we didn't normalize intersections at all, relying instead on normalizing each individual constituent as we broke down the intersection in relationship checking. That worked when `NonNullable<T>` was a conditional type, but not with the new representation.